test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/random/LocalPartGenerator.java

```diff
@@ -0,0 +1,106 @@
+
+import java.util.Random;
+
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * 
+ * The local name part of an email address based on the below rfc <br>
+ * https://tools.ietf.org/html/rfc3696
+ *
+ */
+public class LocalPartGenerator {
+
+    /**
+     * The minimum length of the local part
+     */
+    public static final int MIN_LENGTH = 1;
+    /**
+     * The maximum length of the local part
+     */
+    public static final int MAX_LENGTH = 64;
+    /**
+     * List of valid characters in the local part of the email
+     */
+    public static final char[] LOCALPART_CHARACTERS =
+                    "0123456789!#$%&'*+-/=?^_'.{|}~abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+                                    .toCharArray();
+    /**
+     * Quote are required for the below characters <br>
+     * at-sign ("@"), backslash, double quote, comma, or square brackets
+     */
+    public static final int[] NONSTANDARD_CHARACTERS = new int[] {64, 92, 34, 44, 91, 93};
+
+    /**
+     * Random used for character generation
+     */
+    private final Random random;
+
+    /**
+     * Prevent instantiation outside the package
+     */
+    LocalPartGenerator(final Random random) {
+        this.random = random;
+    }
+
+    /**
+     * Generate the local part
+     * 
+     * @return the local part
+     */
+    public String next() {
+        return generateLocalPart();
+    }
+
+    /**
+     * Generate local part
+     * 
+     * @param text input
+     * @return text to generate
+     */
+    private String generateLocalPart() {
+        String generated = generateStringFromCharacters(random, LOCALPART_CHARACTERS, MIN_LENGTH, MAX_LENGTH);
+        // toss dice
+        if (isRandomlyTrue(6)) {
+            final int positionToInsert = chooseRandomPosition(generated.length());
+            final char nonstandardCharacter = (char) NONSTANDARD_CHARACTERS[chooseRandomPosition(
+                            NONSTANDARD_CHARACTERS.length)];
+            // flip coin
+            final boolean backSlashQuoteType = isRandomlyTrue(2);
+            if (backSlashQuoteType) {
+                generated = String.join("", generated.substring(0, positionToInsert), "\\",
+                                valueOf(nonstandardCharacter),
+                                generated.substring(positionToInsert, generated.length()));
+            } else {
+                generated = String.join("", "\"", generated.substring(0, positionToInsert),
+                                valueOf(nonstandardCharacter),
+                                generated.substring(positionToInsert, generated.length()), "\"");
+            }
+        }
+        if (isValidLocalPart(generated)) {
+            return generated;
+        }
+
+        return generateLocalPart();
+    }
+
+    /**
+     * Check the local part of the email
+     * 
+     * @param textToCheck
+     * @return flag indicating check passed
+     */
+    @VisibleForTesting
```

Comment:
I'm with @allanmckenzie - I don't like private methods being exposed using for testing.